dump.go

```diff
@@ -21,9 +22,30 @@ func withColor(color int, text string) string {
 }
 
 func writeBody(buf *bytes.Buffer, req *http.Request) error {
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		return err
+	var body []byte
+	var err error
+
+	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") {
+		gz, err := gzip.NewReader(req.Body)
```

Comment:
I'm still getting errors, I think this happens when Browser send the Headers `Accept-Encoding: gzip` but the content is not gzipped.

In the attached screen shot you can see that the Info bar is displaying an EOF
![2017-04-10-095833_1599x830_scrot](https://cloud.githubusercontent.com/assets/1288313/24851616/9d1b8f14-1dd4-11e7-8d7c-1147a5ee49a4.png)

I think that case should be handled as well.